Userland/Libraries/LibWeb/HTML/Canvas/CanvasFillStrokeStyles.h

```diff
@@ -23,27 +24,45 @@ class CanvasFillStrokeStyles {
     ~CanvasFillStrokeStyles() = default;
     using FillOrStrokeStyleVariant = Variant<String, JS::Handle<CanvasGradient>, JS::Handle<CanvasPattern>>;
 
-    static CanvasState::FillOrStrokeStyle to_canvas_state_fill_or_stroke_style(auto const& style)
+    void set_fill_style(FillOrStrokeStyleVariant style)
     {
-        return style.visit(
-            [&](String const& string) -> CanvasState::FillOrStrokeStyle {
-                // FIXME: This should parse color strings the same as CSS
-                auto color = Gfx::Color::from_string(string);
-
-                if (!color.has_value())
-                    dbgln_if(CANVAS_RENDERING_CONTEXT_2D_DEBUG, "CanvasFillStrokeStyles: Unsupported canvas fill or stroke style \"{}\". Defaulting to Color::Black.", string);
+        auto& realm = static_cast<IncludingClass&>(*this).realm();
 
-                return color.value_or(Color::Black);
+        // https://html.spec.whatwg.org/multipage/canvas.html#dom-context-2d-fillstyle
+        style.visit(
+            [&](String const& string) {
+                // 1. If the given value is a string, then:
+                // 1. Let context be this's canvas attribute's value, if that is an element; otherwise null.
```

Comment:
Having steps 1 and 1.1 at the same level is a bit confusing, maybe move the first one above the lambda?

Reply:
Good idea!